cmd/service-controller/policy_controller.go

```diff
@@ -24,13 +25,30 @@ import (
 	"github.com/skupperproject/skupper/pkg/event"
 )
 
+var (
+	staticPolicyWatchers []*client.ClusterPolicyValidator
+	policyWatcherLock    sync.Mutex
+)
+
+// AddStaticPolicyWatcher all watchers must be defined before
+// the PolicyController is started
+func AddStaticPolicyWatcher(pv *client.ClusterPolicyValidator) {
+	policyWatcherLock.Lock()
+	defer policyWatcherLock.Unlock()
+	if staticPolicyWatchers == nil {
+		staticPolicyWatchers = []*client.ClusterPolicyValidator{}
+	}
+	staticPolicyWatchers = append(staticPolicyWatchers, pv)
+}
+
 type PolicyController struct {
-	name      string
-	cli       *client.VanClient
-	validator *client.ClusterPolicyValidator
-	informer  cache.SharedIndexInformer
-	queue     workqueue.RateLimitingInterface
-	activeMap map[string]time.Time
+	name             string
+	cli              *client.VanClient
+	validator        *client.ClusterPolicyValidator
+	informer         cache.SharedIndexInformer
+	queue            workqueue.RateLimitingInterface
+	activeMap        map[string]time.Time
+	staticPolicyLock sync.Mutex
```

Comment:
What are the respective roles of this lock and the policyWatcherLock above?

Reply:
Actually I need to get rid of them. They were used in an initial approach I used.
The approach I ended up with no longer needs them, I will remove. Thanks for pointing it out.